### DISCUSSION TITLE
Creating New SSH Key

### DISCUSSION THREAD

**Discussion by michihodges** [community/community#139616] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I can't push from an already setup repo and am required to create a new SSH Key. I've tried this with the id_rsa as well as ed25519. the error is: Key is invalid. You must supply a key in OpenSSH public key format. Please help.

**Comment by DoctorSolo** [comment]
maybe you are using the wrong key

**Reply by michihodges** [reply]
yep

**Comment by davevad93** [comment, marked as answer]
Hi @michihodges ,

You have to add to the GitHub ssh-agent your public key inside the `.pub` file normally stored in `C:\Users\user\.ssh` for Windows users, it's a one-row string that normally starts with `ssh-ed25519`.

For more information on how to generate and add an ssh key to GitHub, take a look at the [docs](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).

Alternatively you can do `git` operations with `HTTPS` by generating a `Personal Access Token` and use it as a password.

 - [Creating a personal access token (classic)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic)

**Reply by Marazali75** [reply]
@DoctorSolo

**Reply by michihodges** [reply]
thanks davevad93 - yep, one-rower

**Reply by davevad93** [reply]
You're welcome @michihodges!👍
